### ISSUE TITLE
Inherited class with overwritten method is not working as advertised in the doc

### ISSUE THREAD

**Issue by MaximeWeyl** [pdoc3/pdoc#75]
### Expected Behavior

The doc says : 

> pdoc considers methods' docstrings inherited from superclass methods', following the normal class inheritance patterns. Consider the following code example:

```
>>> class A:
...     def test(self):
...         """Docstring for A."""
...         pass
...
>>> class B(A):
...     def test(self):
...         pass
...
>>> A.test.__doc__
'Docstring for A.'
>>> B.test.__doc__
None
```

> In Python, the docstring for B.test doesn't exist, even though a docstring was defined for A.test. When pdoc generates documentation for the code such as above, it will automatically attach the docstring for A.test to B.test if B.test does not define its own docstring. In the default HTML template, such inherited docstrings are greyed out.

So I expect to see "test" in the methods of B, documented with the docstring of A.

### Actual Behavior

In my tests with pdoc3 v0.5.0 to v0.6.2, I don't see "test" in the methods of B. It says it is inherited. It is in reality not the case as it is overwritten.

When I add a docstring to B.test, it reappears in the methods of B.

### Steps to Reproduce

1. Create this file main.py : 

```
class A:
    def test(self):
        """Docstring for A."""
        print("From A")


class B(A):
    def test(self):
        print("From B")
```

2. Run pdoc --html
3. Open the generated doc and remark that B.test is not listed in the method, as pdoc thinks it is inherited, which it is not.

### Additional info

- pdoc version: 0.5.0, 0.6.0, 0.6.1, 0.6.2

**Comment by kernc** [comment]
> So I expect to see "test" in the methods of B, documented with the docstring of A.

This will work if you run with `--config show_inherited_members=True`:
https://github.com/pdoc3/pdoc/blob/276f14429fa63f76d7602c0d79825608cc3bf2a8/pdoc/templates/config.mako#L5
The default was [flipped](https://github.com/pdoc3/pdoc/commit/d8602fcf7238de412e8be6e28c31f48f1d858ce3). I guess the documentation could be made to better reflect that ...

**Comment by MaximeWeyl** [comment]
Than you, I did not know about this setting.
It allows to show the docstring of the parent overwritten method.

But I think it is wrong to have the result tagged as "inherited from ...". Only the docstring comes from the superclass method. The method itself is not inherited, as it was overwritten by B. It is very confusing for the final reader who might think that the entire method is inherited.

Are inherited methods and inherited docstrings handled alltogether in pdoc ? If not, we could just change the way it is presented in the final doc.

**Comment by kernc** [comment]
> The method itself is not inherited, as it was overwritten by B.

In [terms of OOP](https://en.wikipedia.org/wiki/Inheritance_(object-oriented_programming)#Overriding), the method itself, the _interface_ is inherited, whereas the method's _implementation_ is overridden.

But with pdoc3 more specifically, these lines handle the inheritance-setting logic:
https://github.com/pdoc3/pdoc/blob/276f14429fa63f76d7602c0d79825608cc3bf2a8/pdoc/__init__.py#L996-L999

So if the subclass' member, or its docstring, is not overridden, the member is marked as inherited from the parent class.

What else would you propose?

**Comment by MaximeWeyl** [comment]
I agree with what you pointed out about interface vs. implementation inheritance.

As a reader of the doc, I would like to know if the implementation was overridden or not (and I personally understand inherited as "not overidden". Maybe I'm wrong here).

As a reader of the doc, I do not really care about whether the docstring was inherited or not. It has no sense for the reader. It is juste a helper for the writer of the doc.


So instead of 

> if the subclass' documented member, or its docstring, is not overridden, the member is marked as inherited from the parent class.

 I would propose :

> If the subclass' documented member is not overridden, the member is marked as inherited from the parent class. If the subclass' documented member is overridden, the member is marked as "overridden" (new mark) from the parent class. And if the subclass' documented member is overridden but has no docstrings, its doctring becomes the docstring of its parent class member.


I think this is what would make most sense to me.
